### PR TITLE
Search by location and add chips

### PR DESCRIPTION
**Pull request by thwalker6**
Resolves # EREGSCSC-1321

**Description-**

We want to make it so that if you search a location ex. 400.200 you will update the filters with the appropriate chips

**This pull request changes...**

- expected change 1

**Steps to manually verify this change...**

1. steps to view and verify change

Type in 400.200 into the search bar.  It should add the chips for part 400 and section 400.200

Clear the results.  Type in 400.200 hello.  No chips should be added but it will search on that as a normal query.

Type in 400.99.  No chips should be added since its an invalid section.  Will just query on 400.99

Type in 500.9.  Same behavior s above

It will work for any valid part/section combination.  It will check the part first to see if the part number is valid, then check the section number to see if it belongs to that part.

### CODE REVIEW

File: solution/ui/regulations/eregs-vite/src/views/Resources.vue [hunk 1]
```diff
@@ -142,7 +138,7 @@ export default {
             },
             supplementalContent: [],
             searchInputValue: "",
-            URL: import.meta.env.VITE_API_URL
+            URL: import.meta.env.VITE_API_URL,
```
Comment (knollfear) [inline solution/ui/regulations/eregs-vite/src/views/Resources.vue:141]: I think this is actually no longer used.

File: solution/ui/regulations/eregs-vite/src/views/Resources.vue [hunk 2]
```diff
@@ -251,27 +258,66 @@ export default {
...
+                });
             }
-            this.$router.push({
-                name: "resources",
-                query: newQueryParams,
-            });
+        },
+
+        async checkPart(payload, scope) {
+            const partExist = this.filters.part.listItems.filter((part) => part.name == payload[0]).length > 0;
```
Comment (knollfear) [inline solution/ui/regulations/eregs-vite/src/views/Resources.vue:314]: .find seems apprpriate here.  you can just replace filter and then lose .length > 0.

File: solution/ui/regulations/eregs-vite/src/views/Resources.vue [hunk 3]
```diff
@@ -251,27 +258,66 @@ export default {
...
+
+            if (partExist && scope === "section") {
+                const sectionList = await getSectionsForPart(42, payload[0]);
+                return (sectionList.filter((section) => section.identifier[1] == payload[1]).length > 0);
```
Comment (knollfear) [inline solution/ui/regulations/eregs-vite/src/views/Resources.vue:318]: find might work here too.

File: solution/ui/regulations/eregs-vite/src/views/Resources.vue [hunk 4]
```diff
@@ -311,71 +363,65 @@ export default {
...
+            this.partDict = newPartDict;
 
             if (dataQueryParams.section) {
-
                 const sections = dataQueryParams.section
                     .split(",")
-                    .filter(x =>
-                        x.match(/^\d+/) && x.match(/\d+$/)
-                    )
+                    .filter((x) => x.match(/^\d+/) && x.match(/\d+$/))
```
Comment (knollfear) [inline solution/ui/regulations/eregs-vite/src/views/Resources.vue:382]: echoing Phil's request for a descriptive variable name.   What is this supposed to be here?

File: solution/ui/regulations/eregs-vite/src/views/Resources.vue [hunk 5]
```diff
@@ -311,71 +363,65 @@ export default {
...
-                })
+                });
             }
             if (dataQueryParams.subpart) {
                 const subparts = dataQueryParams.subpart
                     .split(",")
-                    .filter(x =>
-                        x.match(/^\d+/) && x.match(/\w+$/)
-                    )
+                    .filter((x) => x.match(/^\d+/) && x.match(/\w+$/))
```
Comment (knollfear) [inline solution/ui/regulations/eregs-vite/src/views/Resources.vue:396]: another generic variable name.

**Review by knollfear — COMMENTED** [review]
This looks good.  Just a few small code cleanups and we can get this merged in.

File: solution/ui/regulations/eregs-vite/src/views/Resources.vue (hunk 3 above)
Comment (thwalker6) [reply solution/ui/regulations/eregs-vite/src/views/Resources.vue]: changed

File: solution/ui/regulations/eregs-vite/src/views/Resources.vue (hunk 1 above)
Comment (thwalker6) [reply solution/ui/regulations/eregs-vite/src/views/Resources.vue]: undid the change but merging didnt remove this

**Pull request merged**